packages/react-html/src/serializer.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import {Effect} from '@shopify/react-effect';
+import {Consumer} from './context';
+
+export const EXTRACT_ID = Symbol('serialize');
+
+interface SerializeProps<T> {
+  data(): T;
+}
+
+interface WithSerializedProps<T> {
+  children(data?: T): React.ReactNode;
+}
+
+export function createSerializer<T>(id: string) {
+  function Serialize({data}: SerializeProps<T>) {
+    return (
+      <Consumer>
+        {manager => (
+          <Effect
+            serverOnly
+            kind={EXTRACT_ID}
+            perform={() => manager.set(id, data())}
```

Comment:
:0 did we disable the linting rule against this?!

Reply:
I will disable it, there isn't really a better way to do it here.